addons/account/models/chart_template.py

```diff
@@ -606,6 +610,141 @@ def _setup_utility_bank_accounts(self, template_code, company, template_data):
+            account_template_xml_id = foreign_tax_rep_line['account_id']
+            if existing_accounts.get(account_template_xml_id):
+                return existing_accounts.get(account_template_xml_id)
+
+            sign_comparator = '<' if float(foreign_tax_rep_line.get('factor_percent', 100)) < 0 else '>'
+            minimal_domain = [
+                ('company_id', '=', company.id),
+                ('account_id', '!=', False),
+                ('factor_percent', sign_comparator, 0),
+            ]
+            additional_domain = [
+                ('tax_id.type_tax_use', '=', type_tax_use),
+                ('tax_id.country_id', '=', company.account_fiscal_country_id.id),
+                ('tax_id', 'in', default_company_taxes.ids),
+            ]
+
+            # Trying to find an account being less restrictive on each iteration until the minimum acceptable is reached
+            # If nothing is found, don't fill it to avoid setting a wrong account
+            similar_repartition_line = None
+            while not similar_repartition_line and additional_domain:
+                search_domain = minimal_domain + additional_domain
+                similar_repartition_line = self.env['account.tax.repartition.line'].search(search_domain, limit=1)
+                additional_domain.pop()
+
+            similar_account_id = None
+            if similar_repartition_line:
+                local_tax_account = similar_repartition_line.account_id
+                similar_account_id = create_foreign_tax_account(local_tax_account, _("Foreign tax account (%s)", country.code)).id
+                existing_accounts[account_template_xml_id] = similar_account_id
+
+            return similar_account_id
+
+        def get_adapted_tax_group_data(chart_template_code):
+            tax_groups = self._get_chart_template_data(chart_template_code)['account.tax.group']
+            # replace the foreign accounts with local one created on the fly as close as possible
+            # as what should be done in the local chart template
+            field_and_names = (
+                ('tax_payable_account_id', _("Foreign tax account payable (%s)", country.code)),
+                ('tax_receivable_account_id', _("Foreign tax account receivable (%s)", country.code)),
+                ('advance_tax_payment_account_id', _("Foreign tax account advance payment (%s)", country.code)),
+            )
+            for field, account_name in field_and_names:
+                similar_account_id = None
+                local_tax_group = self.env["account.tax.group"].search([('country_id', '=', company.account_fiscal_country_id.id), (field, '!=', False),])
+                if not local_tax_group:
+                    local_tax_group = self.env["account.tax.group"].search([(field, '!=', False),])
+                if local_tax_group:
+                    similar_account_id = create_foreign_tax_account(local_tax_group[field], account_name).id
```

Comment:
What's wrong with this?
```suggestion
            local_tax_groups = self.env["account.tax.group"].search([('country_id', '=', company.account_fiscal_country_id.id)])
            for field, account_name in field_and_names:
                similar_account_id = create_foreign_tax_account(local_tax_groups[field][:1], account_name).id
```

Reply:
I also think that you are creating way too many accounts here with your loop, you should have a cache based on `existing_account` in `create_foreign_tax_account`

Reply:
I added the caching on xml_id